### PR TITLE
Implemented logic to differentiate between permission errors and atte…

### PR DESCRIPTION
**Pull request by hokiegeek2**
…mpting opening of non-HDF5 files, added logic and ErrorWithContext to handle any other HDF5 file format errors, added error message generation logic for the load_all workflows, added io_util error handling unit tests #519

### CODE REVIEW

**Comment by mhmerrill** [comment]
> …mpting opening of non-HDF5 files, added logic and ErrorWithContext to handle any other HDF5 file format errors, added error message generation logic for the load_all workflows, added io_util error handling unit tests #519

File: src/GenSymIO.chpl [hunk 1]
```diff
@@ -981,6 +1055,7 @@ module GenSymIO {
                 }
             }
         } catch e: FileNotFoundError {
+              writeln("HELLO");
```
Comment (reuster986) [inline src/GenSymIO.chpl:1058]: Is this a debug message?

**Review by reuster986 — APPROVED** [review]
I like it!

File: src/GenSymIO.chpl (hunk 1 above)
Comment (hokiegeek2) [reply src/GenSymIO.chpl:1058]: ROTFL, sure is! Cleanup in aisle 5 is needed on this!

**Comment by hokiegeek2** [comment]
> I like it!

Cool, thanks @reuster986, just trying to knock out a few things that neither you nor @mhmerrill have had a chance to work on.

File: src/GenSymIO.chpl (hunk 1 above)
Comment (hokiegeek2) [reply src/GenSymIO.chpl]: fixed

**Review by mhmerrill — APPROVED** [review]
this looks ok to me, it looks like Bill approved, i'll merge when CI finishes.

**Pull request merged**